sdk/language-support/ts/daml-types/index.ts

```diff
@@ -245,19 +245,31 @@ const registeredTemplates: { [key: string]: Template<object> } = {};
  */
 export const registerTemplate = <T extends object>(
   template: Template<T>,
+  packageSpecifiers: string[] = [],
 ): void => {
-  const templateId = template.templateId;
-  const oldTemplate = registeredTemplates[templateId];
-  if (oldTemplate === undefined) {
-    registeredTemplates[templateId] = template as unknown as Template<
-      object,
-      unknown,
-      string
-    >;
-    console.debug(`Registered template ${templateId}.`);
-  } else {
-    console.warn(`Trying to re-register template ${templateId}.`);
-  }
+  const aliases = packageSpecifiers.map(templateIdWithPackageId(template));
+  new Set([template.templateId].concat(aliases)).forEach(tmplId => {
+    if (tmplId in registeredTemplates) {
+      console.warn(`Trying to re-register template ${tmplId}.`);
+    } else {
+      registeredTemplates[tmplId] = template as unknown as Template<
+        object,
+        unknown,
+        string
+      >;
+      console.debug(`Registered template ${tmplId}.`);
+    }
+  });
+};
+
+/**
+ * @internal
+ */
+export const templateIdWithPackageId = <T extends object, K, I extends string>(
```

Comment:
New curried function which replaces the package specifier at the beginning of a template id. This is used when registring templates, and also useful in tests to create expectations on the templateId's returned (which will use a package id), if the request templateId used a package name.